src/legacy/core_plugins/dashboard_embeddable_container/public/embeddable/grid/dashboard_grid.tsx

```diff
@@ -0,0 +1,294 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { injectI18n } from '@kbn/i18n/react';
+import classNames from 'classnames';
+import _ from 'lodash';
+import React from 'react';
+import ReactGridLayout, { Layout } from 'react-grid-layout';
+import 'react-grid-layout/css/styles.css';
+import 'react-resizable/css/styles.css';
+
+// @ts-ignore
+import sizeMe from 'react-sizeme';
+import { toastNotifications } from 'ui/notify';
+import { Subscription } from 'rxjs';
+import { DashboardConstants } from '../../../../kibana/public/dashboard/dashboard_constants';
+import { ViewMode, EmbeddableChildPanel } from '../../../../embeddable_api/public';
+import { DASHBOARD_GRID_COLUMN_COUNT, DASHBOARD_GRID_HEIGHT } from '../dashboard_constants';
+import { DashboardContainer } from '../dashboard_container';
+import { DashboardPanelState, GridData } from '../types';
+
+let lastValidGridSize = 0;
+
+/**
+ * This is a fix for a bug that stopped the browser window from automatically scrolling down when panels were made
+ * taller than the current grid.
+ * see https://github.com/elastic/kibana/issues/14710.
+ */
+function ensureWindowScrollsToBottom(event: { clientY: number; pageY: number }) {
+  // The buffer is to handle the case where the browser is maximized and it's impossible for the mouse to move below
+  // the screen, out of the window.  see https://github.com/elastic/kibana/issues/14737
+  const WINDOW_BUFFER = 10;
+  if (event.clientY > window.innerHeight - WINDOW_BUFFER) {
+    window.scrollTo(0, event.pageY + WINDOW_BUFFER - window.innerHeight);
+  }
+}
+
+function ResponsiveGrid({
+  size,
+  isViewMode,
+  layout,
+  onLayoutChange,
+  children,
+  maximizedPanelId,
+  useMargins,
+}: {
+  size: { width: number };
+  isViewMode: boolean;
+  layout: Layout[];
+  onLayoutChange: () => void;
+  children: JSX.Element[];
+  maximizedPanelId: string;
+  useMargins: boolean;
+}) {
+  // This is to prevent a bug where view mode changes when the panel is expanded.  View mode changes will trigger
```

Comment:
Not a complete copy from the original `dashboard_grid.tsx`, some code was removed that I think is no longer needed with the newer styles, like the old onBlur, onFocus handling, and the in-grid panel migrations away from older style data is moved out, all migrations should have occurred before we get here, and data be in the shape we expect.